tsfc/driver.py

```diff
@@ -349,3 +354,12 @@ def lower_integral_type(fiat_cell, integral_type):
         entity_ids = list(range(len(fiat_cell.get_topology()[integration_dim])))
 
     return integration_dim, entity_ids
+
+
+def pick_mode(mode):
+    "Return one of the specialized optimisation modules from a mode string."
+    if mode == "vanilla":
+        import tsfc.vanilla as m
```

Comment:
I do not envisage composing modes as such, strictly speaking they are merely meant for configuration. It does make sense to compose the underlying transformations, however. Unrolling `IndexSum`s is not really implemented by the vanilla mode, but by `gem.optimise.unroll_indexsum`. Any mode can just call this function if needed.